### PR TITLE
Update CHANGELOG for 0.12.3

### PR DESCRIPTION
**Pull request by evan2645**
Reference https://github.com/spiffe/spire/pull/2264

Signed-off-by: Evan Gilman <egilman@vmware.com>

### CODE REVIEW

**Review by azdagron — DISMISSED** [review]
:+1:

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,20 @@
 # Changelog
 
+## [0.12.3] - 2021-05-14
+
+### Added
+- The `k8s-workload-registrar` now supports federation (#2160)
+- The `k8s_bundle` notifier plugin can now keep API service CA bundles up to date
```
Comment (amartinezfayo) [inline CHANGELOG.md:7]: Missing PR#

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,20 @@
...
+- SPIRE Server internal cache reload timing can now be tuned (experimental) (#2169)
+
+### Changed
+- Prometheus metrics that are emitted infrequently will no longer disappear after emission (#2239)
+- The `k8s-workload-registrar` now uses paging to support very large deployments of 10,000+ pods
```
Comment (amartinezfayo) [inline CHANGELOG.md:12]: Missing PR#

**Review by amartinezfayo — COMMENTED** [review]
Looking great! Just missing a couple of PR#.

**Comment by evan2645** [comment]
🤦 thank you @amartinezfayo! I fixed it, and I also made a slight wording change on the agent selector bugfix s/new agents/newly attested agents/

**Review by azdagron — APPROVED** [review]
\o/

**Review by amartinezfayo — APPROVED** [review]
🚀

**Pull request merged**